axiom/tests/test_segment_private.c

```diff
@@ -328,6 +370,19 @@ static void test_destroy_typed_attributes(void) {
   nr_segment_destroy_typed_attributes(NR_SEGMENT_DATASTORE, NULL);
   nr_segment_destroy_typed_attributes(NR_SEGMENT_DATASTORE,
                                       &s.typed_attributes);
+  nr_segment_destroy_typed_attributes(NR_SEGMENT_MESSAGE, NULL);
+  nr_segment_destroy_typed_attributes(NR_SEGMENT_MESSAGE, &s.typed_attributes);
+
+  /*
+   * Test : Clean up typed attributes for a message segment
+   */
+  s.type = NR_SEGMENT_MESSAGE;
+  s.typed_attributes = nr_zalloc(sizeof(nr_segment_typed_attributes_t));
+  s.typed_attributes->message.destination_name = nr_strdup("queue_name");
+  s.typed_attributes->message.messaging_system = nr_strdup("aws_sqs");
+  s.typed_attributes->message.server_address = nr_strdup("localhost");
+
+  nr_segment_destroy_typed_attributes(NR_SEGMENT_MESSAGE, &s.typed_attributes);
```

Comment:
Does this need a test to verify the typed attributes were cleaned up?

Reply:
As with other tests, valgrind is used to verify proper mem cleanup.
Comment updates to indicate that.
https://github.com/newrelic/newrelic-php-agent/commit/fde6d510b065a30bc0b242411b5a39a31b86da6a

Reply:
I'm thinking more about the state of the pointers being set to a known value like `NULL`.  I am not sure valgrind would evaluate if this is happening.

Reply:
All of these use nr_free which sets to null, purpose of this isn't to test nr_free, just verify it's been called.

Reply:
My thought is the test is to verify `nr_segment_destroy_typed_attributes()` does these things properly - we shouldn't assume what it is using internally but treat it as an unknown and verify the state after calling it meets our expectations.  Maybe you can infer it called `nr_free()` if valgrind doesn't complain about leaks.  My preference is that the unit test has coverage of all these things and valgrind is just a backup for memory management issues.

Reply:
Please see: [0670292](https://github.com/newrelic/newrelic-php-agent/pull/991/commits/0670292ebb5d1562c4e7f43518bdeaf59e8a8186)